libconfig/src/config_parser.c

```diff
@@ -447,35 +447,48 @@ bool config_parser_add_key_values(config_parser_type * config,
                                   const char * config_filename,
                                   config_schema_unrecognized_enum unrecognized)
 {
-  if (hash_has_key(config->messages, kw))
-    printf("%s \n", (const char *) hash_get(config->messages, kw));
-
   if (!config_has_schema_item(config, kw)) {
-    if (unrecognized == CONFIG_UNRECOGNIZED_WARN)
-      fprintf(stderr,
-              "** Warning keyword: %s not recognized when parsing: %s --- \n",
+
+    if (unrecognized == CONFIG_UNRECOGNIZED_IGNORE)
+      return false;
+
+    if (unrecognized == CONFIG_UNRECOGNIZED_WARN) {
+      fprintf(stderr, "** Warning keyword: %s not recognized when parsing: %s --- \n",
```

Comment:
Should we log this?

Reply:
The config parser is also used for other tasks where logging to the shared log facility does not make sense  - so I would say no?

Reply:
You say no, I say merge.